Sources/HttpServer/Router/Router.swift

```diff
@@ -21,6 +21,13 @@ public class Router {
 
         if let validRoute = route {
             perform_action(request: request, route: validRoute)
+            if needsAuthorization(route: validRoute) {
+                if authorizeSucceeded(route: validRoute, request: request) {
+                    return responseGenerator.generate200Response(method: requestMethod, url: requestUrl)
+                } else {
+                    return responseGenerator.generate401Response(realm: validRoute.getRealm())
+                }
+            }
```

Comment:
This method is pretty nasty - three nested if statements! I would refactor this a bit.